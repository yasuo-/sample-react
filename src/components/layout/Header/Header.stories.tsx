import React from 'react';

import { ComponentStory, ComponentMeta } from '@storybook/react';

import { Header } from './Header';

export default {
  title: 'Components/Layout/Header',
  component: Header,
  parameters: {
    layout: 'fullscreen',
  },
} as ComponentMeta<typeof Header>;

const Template: ComponentStory<typeof Header> = (args) => <Header {...args} />;

export const Default = Template.bind({});

Default.args = {
  siteTitle: 'logo',
  src: 'https://via.placeholder.com/150',
  logoWidth: 150,
  logoHeight: 50,
};
