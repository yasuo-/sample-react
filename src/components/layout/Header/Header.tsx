import React from 'react';

import { Link } from 'react-router-dom';

import { PATH } from '@/routes/path';

type HeaderProps = {
  siteTitle?: string;
  src?: string;
  logoWidth?: number;
  logoHeight?: number;
};

/**
 * Header
 * @param alt
 * @param src
 * @param logoWidth
 * @param logoHeight
 * @returns Header
 */
export const Header: React.FC<HeaderProps> = ({
  siteTitle = 'logo',
  src = 'https://via.placeholder.com/150',
  logoWidth = 150,
  logoHeight = 50,
}) => {
  return (
    <header className="w-full bg-fuego-500">
      <div className="flex justify-center items-center h-10">
        <h1>
          <Link to={PATH.HOME} className="hover:opacity-75">
            <img src={src} width={logoWidth} height={logoHeight} alt={siteTitle} />
          </Link>
        </h1>
      </div>
    </header>
  );
};

Header.displayName = 'Header';
