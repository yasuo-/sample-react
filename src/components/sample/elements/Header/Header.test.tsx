import React from 'react';

import { fireEvent, render, screen, waitFor } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';

import { Header } from './Header';

/**
 * renderWithRouter
 * helper function to render component with router
 * @param component
 */
const renderWithRouter = (component: React.ReactNode) => {
  return {
    ...render(<MemoryRouter>{component}</MemoryRouter>),
  };
};

describe('Header Element', () => {
  it('displayName is Header', () => {
    expect(Header.displayName).toBe('Header');
  });

  it('should take a snapshot', () => {
    const { asFragment } = renderWithRouter(<Header siteTitle={'test title'} />);
    expect(asFragment).toMatchSnapshot();
  });

  it('renders test title', () => {
    renderWithRouter(<Header siteTitle={'test title'} />);
    const screenElement = screen.getByText(/test title/i);
    expect(screenElement).toBeInTheDocument();
  });

  it('Should Link Click Move Google', () => {
    renderWithRouter(<Header siteTitle={'test title'} to={'https://www.google.com'} />);
    fireEvent.click(screen.getByText(/test title/i));

    waitFor(() => {
      expect(window.location.href).toBe('https://www.google.com');
    });
  });
});
