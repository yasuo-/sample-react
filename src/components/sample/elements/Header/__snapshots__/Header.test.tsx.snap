// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Element should take a snapshot 1`] = `[Function]`;
